packages/admin/cms-admin/src/common/MasterMenu.tsx

```diff
@@ -20,10 +24,12 @@ export function getMenuFromMasterMenuData(items: MasterMenuData): MenuItem[] {
                 to: to ?? route?.path?.toString() ?? "",
             },
             hasSubmenu: !!submenu,
-            submenu: submenu ? submenu.map(mapFn) : [],
+            submenu: submenu ? submenu.filter(filterFn).map(mapFn) : [],
         };
     };
-    return items.map(mapFn);
+    const filterFn = (item: MasterMenuItem): boolean =>
+        !item.requiredPermission || (context !== undefined && context.isAllowed(context.currentUser, item.requiredPermission));
```

Comment:
Should we throw an error if `item.requiredPermission` is set, but no `context` is defined?

Reply:
d46b5fa4bbc8a34ae912c7f4248bc4954f9b9983